cases_test/test_2d_free_stream_around_cylinder/src/2d_free_stream_around_cylinder .cpp

```diff
@@ -0,0 +1,275 @@
+	system.run_particle_relaxation_ = false;
+	/** Tag for computation start with relaxed body fitted particles distribution. */
+	system.reload_particles_ = false;
+	/** Tag for computation from restart files. 0: start with initial condition. */
+	system.restart_step_ = 0;
+
+	/** handle command line arguments. */
+	system.handleCommandlineOptions(ac, av);
+	In_Output in_output(system);
+
+	/**
+	 * @brief Creating body, materials and particles for a water block.
+	 */
+	WaterBlock* water_block = new WaterBlock(system, "WaterBody");
+	water_block->setBodyDomainBounds(fluid_body_domain_bounds);
+	WaterMaterial* water_material = new WaterMaterial();
+	FluidParticles fluid_particles(water_block, water_material);
+	/**
+	 * @brief 	Creating body, materials and particles for solid cylinder.
+	 */
+	Cylinder* cylinder = new Cylinder(system, "Cylinder");
+	if (!system.run_particle_relaxation_ && system.reload_particles_) cylinder->useParticleGeneratorReload();
+	SolidParticles cylinder_particles(cylinder);
+	/**
+	 * @brief 	Particle and body creation of beam and fluid observers.
+	 */
+	FluidObserver* fluid_observer = new FluidObserver(system, "FluidObserver");
+	BaseParticles flow_observer_particles(fluid_observer);
+	/**
+	 * @brief define simple data file input and outputs functions.
+	 */
+	BodyStatesRecordingToVtu write_real_body_states(in_output, system.real_bodies_);
+	RestartIO restart_io(in_output, system.real_bodies_);
+	/** topology */
+	BodyRelationInner* water_block_inner = new BodyRelationInner(water_block);
+	ComplexBodyRelation* water_block_complex = new ComplexBodyRelation(water_block_inner, { cylinder });
+	BodyRelationContact* cylinder_contact = new BodyRelationContact(cylinder, { water_block });
+	BodyRelationContact* fluid_observer_contact = new BodyRelationContact(fluid_observer, { water_block });
+
+	/** check whether run particle relaxation for body fitted particle distribution. */
+	if (system.run_particle_relaxation_)
+	{
+		/** body topology only for particle relaxation */
+		BodyRelationInner* cylinder_inner = new BodyRelationInner(cylinder);
+		/**
+		 * @brief 	Methods used for particle relaxation.
+		 */
+		 /** Random reset the insert body particle position. */
+		RandomizePartilePosition random_inserted_body_particles(cylinder);
+		/** Write the body state to Vtu file. */
+		BodyStatesRecordingToVtu write_inserted_body_to_vtu(in_output, { cylinder });
+		/** Write the particle reload files. */
+		ReloadParticleIO write_particle_reload_files(in_output, { cylinder });
+
+		/** A  Physics relaxation step. */
+		relax_dynamics::RelaxationStepInner relaxation_step_inner(cylinder_inner);
+		/**
+		  * @brief 	Particle relaxation starts here.
+		  */
+		random_inserted_body_particles.parallel_exec(0.25);
+		relaxation_step_inner.surface_bounding_.parallel_exec();
+		write_real_body_states.writeToFile(0.0);
+
+		/** relax particles of the insert body. */
+		int ite_p = 0;
+		while (ite_p < 1000)
```

Comment:
is this 1000 depending on certain conditions?

Reply:
No. It is a large arbitrary big number and will not change the final results much.

Reply:
Can we set a condition here instead of a number? I think it works most of the time, but it can happen that it's not enough